merging_iter.go

```diff
@@ -601,10 +610,18 @@ func (m *mergingIter) isPrevEntryDeleted(item *mergingIterItem) bool {
 				// We can seek up to the max of smallestUserKey and tombstone.Start.UserKey.
 				//
 				// Using example 1 above, we can seek to the larger of c and b, which is c.
+				//
+				// Progress argument: We know that the iterator in this file is positioned within
+				// its bounds and at a key X < item.key (otherwise it would be the max of the heap).
+				// So smallestUserKey <= item.key.UserKey and we already know that
+				// l.tombstone.Start.UserKey <= item.key.UserKey. So the seekKey computed below
+				// is <= item.key.UserKey, and since we do a seekLT() we will make backwards
+				// progress.
 				seekKey := l.tombstone.Start.UserKey
 				if l.smallestUserKey != nil && m.heap.cmp(l.smallestUserKey, seekKey) > 0 {
 					seekKey = l.smallestUserKey
 				}
+				// we know that seekKey <= item.key.UserKey
```

Comment:
Nit: comments which are on their own line (I.e not sharing a line with code) should be properly capitalized and punctuated.